src/main/java/com/puppycrawl/tools/checkstyle/xpath/ElementNode.java

```diff
@@ -0,0 +1,97 @@
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import net.sf.saxon.type.Type;
+
+/**
+ * Represents element node of Xpath-tree.
+ *
+ * @author Timur Tibeyev
+ */
+public class ElementNode extends AbstractNode {
+    /** Represents value of TokenTypes#IDENT. */
+    private String image;
+
+    /**
+     * Creates a new {@code ElementNode} instance.
+     *
+     * @param tokenType the number representing constant from {@code TokenTypes}
+     * @param parent {@code Node} parent of the current node
+     * @param document {@code Node} root of the tree
+     * @param detailNode reference to {@code DetailNode}
+     * @param detailAST reference to {@code DetailAST}
+     */
+    public ElementNode(int tokenType, Node parent, Node document, DetailNode detailNode, DetailAST detailAST) {
+        super(tokenType);
+        this.parent = parent;
+        this.document = document;
+        this.detailNode = detailNode;
+        this.detailAST = detailAST;
+
+        if (detailAST.branchContains(TokenTypes.IDENT)) {
```

Comment:
it is not hack, as I understood `IDENT` is used to identify names of classes, methods, parameters and etc.
So this part of code extracts name of the current AST element and make it possible to query like `@text = 'SomeName' ` using `getAttribute(namespace, localpart)` method